crates/fuel-core/src/schema/node_info.rs

```diff
@@ -204,22 +203,59 @@ impl TxPoolStats {
 }
 
 #[derive(Clone)]
-struct Indexation(HashSet<IndexationKind>);
+struct Indexation(u8);
+
+impl Indexation {
+    pub fn new() -> Self {
+        Self(0)
+    }
+
+    pub fn contains(&self, kind: &IndexationKind) -> bool {
+        self.0 & (1 << *kind as u8) != 0
```

Comment:
Do you think we can do this is a more readable way?

Reply:
Mhhh we can highly comment it on the top of the function but this is the only one I'm used to do it and always do it like that so I don't know any other way.

Reply:
Maybe you can remind me why we prefer `u8` over `HashSet` [here](https://github.com/FuelLabs/fuel-core/pull/2604/files#diff-8d14f23f382a31131da03da9061317cef7d129f65322d79dd750bc20445506cbR206)?

Reply:
Less memory, less algorithmic complexity

Reply:
I would rather not trade code readability over minor performance gains, but it's just me :-)

Reply:
Personally, I don't see any code readibility issue here, it's a pretty standard pattern but I can understand. This is just a proposition, you can refuse it I'm fine with it.
However, I think the initialization compile-time safe should be ported to your solution because it provide safety measures for the future.

Personal note, I don't think it's "minor performance gains" has we are working on an API that could be triggered thousands of times per seconds but it's a non-sense discussion without real results so I'm fine not accepting this.

Reply:
Feel free to add what ever you want from this on your PR and I will close this one.

Reply:
No, I think I'll get this as a whole, seeing @xgreenx being ok with that. Not a big deal ;)
Thanks for implementing this :+1: